client/functions.lua

```diff
@@ -44,12 +45,22 @@ end
 
 exports('HasKeys', public.hasKeys)
 
-function public.toggleEngine()
-    local vehicle = cache.vehicle
-    if vehicle and public.hasKeys(qbx.getVehiclePlate(vehicle)) then
-        local engineOn = GetIsVehicleEngineRunning(vehicle)
-        SetVehicleEngineOn(vehicle, not engineOn, false, true)
-    end
+---Checks if player has vehicle keys of or access to the vehicle is provided as part of his job.
+---@param vehicle number The entity number of the vehicle.
+---@param plate string? The plate number of the vehicle.
+---@return boolean? `true` if player has access to the vehicle, `nil` otherwise.
+function public.getIsVehicleAccessible(vehicle, plate)
+    plate = plate or qbx.getVehiclePlate(vehicle)
```

Comment:
Can't plate be derived from vehicle?

Reply:
This is only an optimization when you already have a registration number in variable

Reply:
Hmmm ok. I'm not convinced that the performance gained from not executing the native is worth the complication to the API. What if the plate passed is not the vehicle's plate? Is the native execution performance worth optimizing, or is it negligible?

Reply:
I added this argument only because I already had the license plate number in a variable and the function is used in a loop, the stress of which I wanted to reduce as much as possible.

Reply:
I haven't verified what effect this has on performance